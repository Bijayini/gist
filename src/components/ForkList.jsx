import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import theme from '../lib/theme';

const ForkList = ({ forks }) => {
  return (
    <Fragment>
      <div className="fork-container">
        {forks.map(fork => (
          <div className="avatar" key={fork}>
            <img src={fork} alt="image" className="fork" />
          </div>
        ))}
      </div>
      <style jsx="true">
        {`
          .fork-container {
            display: flex;
          }
          .avatar {
            background: ${theme.grey1};
            margin-right: 5px;
            width: 30px;
            height: 30px;
          }
          .fork {
            max-width: 100%;
            border-radius: 3px;
          }
        `}
      </style>
    </Fragment>
  );
};

ForkList.propTypes = {
  forks: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default ForkList;
