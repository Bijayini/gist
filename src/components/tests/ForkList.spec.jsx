import React from 'react';
import { shallow } from 'enzyme';

import ForkList from '../ForkList';

describe('<ForkList />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      forks: ['some value1', 'some value2'],
    };
    component = shallow(<ForkList {...props} />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
