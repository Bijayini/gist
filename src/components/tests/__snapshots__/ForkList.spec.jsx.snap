// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForkList /> should match with snapshot 1`] = `
"<Fragment>
  <div className=\\"jsx-2124877363 fork-container\\">
    <div className=\\"jsx-2124877363 avatar\\">
      <img src=\\"some value1\\" alt=\\"image\\" className=\\"jsx-2124877363 fork\\" />
    </div>
    <div className=\\"jsx-2124877363 avatar\\">
      <img src=\\"some value2\\" alt=\\"image\\" className=\\"jsx-2124877363 fork\\" />
    </div>
  </div>
  <JSXStyle id=\\"52773119\\" dynamic={{...}}>
    .fork-container.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.avatar.__jsx-style-dynamic-selector{background:#4a4a4a;margin-right:5px;width:30px;height:30px;}.fork.__jsx-style-dynamic-selector{max-width:100%;border-radius:3px;}
  </JSXStyle>
</Fragment>"
`;
